scripts/gdal/gdal_bands.py

```diff
@@ -36,31 +40,49 @@ def get_gdal_band_offset(file: str, info: Optional[GdalInfo] = None) -> List[str
 
     bands = info["bands"]
 
-    alpha_band = find_band(bands, "Alpha")
-    alpha_band_info: List[str] = []
-    if alpha_band:
-        alpha_band_info.extend(["-b", str(alpha_band["band"])])
+    band_alpha_arg: List[str] = []
+    if band_alpha := find_band(bands, "Alpha"):
+        band_alpha_arg = ["-b", str(band_alpha["band"])]
 
     # Grey scale imagery, set R,G and B to just the grey_band
-    grey_band = find_band(bands, "Gray")
-    if grey_band:
-        grey_band_index = str(grey_band["band"])
-        return ["-b", grey_band_index, "-b", grey_band_index, "-b", grey_band_index] + alpha_band_info
+    if band_grey := find_band(bands, "Gray"):
+        band_grey_index = str(band_grey["band"])
+        return ["-b", band_grey_index, "-b", band_grey_index, "-b", band_grey_index] + band_alpha_arg
+
+    if band_palette := find_band(bands, "Palette"):
+        if colour_table := band_palette["colorTable"]:
+            palette_channels = len(colour_table["entries"][0])
+            if palette_channels == 4:
+                return ["-expand", "rgba"]
+            if palette_channels == 3:
+                return ["-expand", "rgb"]
+            get_log().error(
```

Comment:
In the case of logging this error message, you'll also log the error message above, out of the `if`. You'll have both `unknown_palette_band_type` and `palette_band_missing_colorTable`. Is that the expected behaviour?